File: Algorithms/searching/binary_search.py

def binary_search(arr, target, first= 0, last=-1):
    if last==-1: last= len(arr)

    while last> first:
        middle= first+ (last- first)/2

        if arr[middle]== target: return middle
        elif arr[middle]< target: first= middle+1
        else: last= middle

    return None

File: Algorithms/searching/binary_search.java
public static int binary_search(int[] array, int k){
    int first= 0;
    int last= array.size()-1;

    while(first<= last){
        int middle= first+ (last- first)/2;

        if(array[middle]< k) first= middle+1;
        else if(array[middle]> k) last= middle- 1;
        else return middle;
    }

    return -1;
}